src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -194,4 +200,24 @@ private function mapRequestPayload(Request $request, string $type, MapRequestPay
             throw new HttpException(Response::HTTP_BAD_REQUEST, sprintf('Request payload contains invalid "%s" data.', $format), $e);
         }
     }
+
+    private function mapRequestHeader(Request $request, string $type, MapRequestHeader $attribute): ?object
+    {
+        $headers = [];
+        $requestHeaders = $request->headers->all();
+
+        array_walk($requestHeaders, function ($value, $key) use (&$headers) {
+            $key = $this->kebabCaseToCamelCase($key);
+            $headers[$key] = $value[0];
+        });
+
+        return $this->serializer->denormalize($headers, $type, null, self::CONTEXT_DESERIALIZE + $attribute->serializationContext);
+    }
+
+    private function kebabCaseToCamelCase($string): string
```

Comment:
I am not sure this is the best place to actually put this, you might be able to do it inline or maybe there is a helper function somewhere in the source code you could use?